src/cplscheme/CouplingData.cpp

```diff
@@ -36,16 +36,12 @@ int CouplingData::getDimensions() const
 
 int CouplingData::getSize() const
 {
-  // @todo this correct implementation breaks a ton of tests that don't define vertices of a test mesh
-  //return _mesh->nVertices() * getDimensions();
-  return sample().values.size();
+  return _mesh->nVertices() * getDimensions();
```

Comment:
```suggestion
  const int size = _mesh->nVertices() * getDimensions();
  PRECICE_ASSERT(sample().values.size() ==  size);
  return size;
```
Similar in the change below. Generally all this should be consistent, no?

Reply:
When is the `sample()` created though?

Reply:
Hmm. True. We should keep this until we remove the sample member.

Reply:
Actually, `sample()` is a better place for this check.